gateway/mw_external_oauth.go

```diff
@@ -51,7 +58,7 @@ func (k *ExternalOAuthMiddleware) ProcessRequest(w http.ResponseWriter, r *http.
 	provider := k.Spec.ExternalOAuth.Providers[0]
 
 	if provider.JWT.Enabled {
-		valid, identifier, err = k.jwt(token)
+		valid, identifier, err = k.jwt(token, provider.JWT)
```

Comment:
can we remove the parameter and access to jwt inside `jwt` function with `k.Spec`?

Reply:
This might be valid at this point (when there is only one provider, or we only consider 1st entry in the array). Later on when we start considering multiple fallback providers, I think this parameter would be needed

Reply:
Later we may pass index as a param.